pkg/store/bucket.go

```diff
@@ -2340,9 +2340,9 @@ func (r *bucketIndexReader) fetchPostings(ctx context.Context, keys []labels.Lab
 				l   index.Postings
 				err error
 			)
-			if isDiffVarintSnappyEncodedPostings(b) {
+			if isDiffVarintSnappyEncodedPostings(b) || isDiffVarintSnappyStreamedEncodedPostings(b) {
 				s := time.Now()
-				clPostings, err := diffVarintSnappyDecode(b)
+				clPostings, err := getDecodingFunction(b)(b)
```

Comment:
I find the `(b)(b)` a bit confusing, what do you think if we simply did something like `decodePostings(b)` and call the particular decoding function inside of _that_ function and just return the result + err?